node/src/GlideClusterClient.ts

```diff
@@ -314,6 +318,141 @@ export class GlideClusterClient extends BaseClient {
         );
     }
 
+    /**
+     * @internal
+     */
+    protected scanOptionsToProto(
+        cursor: string,
+        options?: ScanOptions,
+    ): command_request.ClusterScan {
+        const command = command_request.ClusterScan.create();
+        command.cursor = cursor;
+
+        if (options?.match) {
+            command.matchPattern = Buffer.from(options.match);
+        }
+
+        if (options?.count) {
+            command.count = options.count;
+        }
+
+        if (options?.type) {
+            command.objectType = options.type;
+        }
+
+        return command;
+    }
+
+    /**
+     * @internal
+     */
+    protected createClusterScanPromise(
+        cursor: ClusterScanCursor,
+        options?: ScanOptions & DecoderOption,
+    ): Promise<[ClusterScanCursor, GlideString[]]> {
+        // separate decoder option from scan options
+        const { decoder, ...scanOptions } = options || {};
+        const stringDecoder: boolean =
+            (decoder ?? this.defaultDecoder) === Decoder.String;
+
+        if (this.isClosed) {
+            throw new ClosingError(
+                "Unable to execute requests; the client is closed. Please create a new client.",
+            );
+        }
```

Comment:
Could be a common part into superclass

Reply:
I think it's complicating and adding “generalization” for single case.
There's no other cases when we create different command Promise.
Currently, its YAGNI.
When more “non-command” API will be added, it will be worth refactoring.